### DISCUSSION TITLE
How is job process locking handled?

### DISCUSSION THREAD

**Discussion by lovgrandma** [bee-queue/bee-queue#860] [Q&A]
I am running this on a node.js API and we have recently clusterized the application.

It seems to be working fine without duplicating on multiple workers but Im wondering the work that was put in for locking of jobs? If I can see code or get a little bit more insight into the thought process behind this?

Our jobs are created via api call thats sent to a single worker but since its in the same redis cache when createJob is run I am guessing any of the workers can potentially process it. So Im wondering about that locking moment before job is picked up.

It _looks_ like it is working but I want to be sure of theory behind this.

**Comment by lovgrandma** [comment]
I am guessing that if each id is different here then there is no duplication across workers?

We also have appending of worker id beside all logs as well.
I know this sounds like a dumb question but we are about to containerize and put on k8s so I want to be sure

```
queue.process(async (job, done) => {
    await limiter.schedule(async () => { // Limiter will control the amount of jobs pushed through
        console.log('Processing job', job.id)
        handleProcessJobs(job, done)
    })
})
```

**Comment by compwright** [comment]
Redis supports atomic operations, and my understanding is bee-queue leverages this feature along with Lua scripts to accomplish this. You should be fine, but of course, test it.
